docs/USAGE-GUIDE.md

```diff
@@ -1401,6 +1401,23 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
 **s2n_client_hello_get_extension_length** returns the number of bytes the given extension type takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
 **s2n_client_hello_get_extension_by_id** copies into the **out** buffer **max_length** bytes of a given extension type on the ClientHello and returns the number of copied bytes.
 
+### s2n\_client\_hello\_get\_session\_id
+
+```c
+int s2n_client_hello_get_session_id_length(struct s2n_client_hello *ch, uint32_t *out_length);
+int s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, uint32_t *out_length, uint32_t max_length);
+```
+
+- **ch** The s2n_client_hello on the s2n_connection. The handle can be obtained using **s2n_connection_get_client_hello**.
+- **out** Pointer to a buffer into which the session id bytes should be copied.
+- **out_length** Pointer to a **uint32_t** which for **s2n_client_hello_get_session_id_length** will be used to store the full length of session_id and for **s2n_client_hello_get_session_id** will be used to store the number of bytes copied to the **out** buffer.
+- **max_length** Max number of bytes to copy into the **out** buffer.
+
+**s2n_client_hello_get_session_id_length** in case of success returns **S2N_SUCCESS** and stores in **out_length** the length of the ClientHello session id in bytes; in case of failure returns **S2N_FAILURE** with no guarantees around **out_length** content. The **out_length** can be used to allocate the **out** buffer for the **s2n_client_hello_get_session_id** call.
+**s2n_client_hello_get_session_id** in case of success returns **S2N_SUCCESS**, copies up to **max_length** bytes of a ClientHello session_id into the **out** buffer and stores the number of copied bytes in **out_length**; in case of failure returns **S2N_FAILURE** with no guarantees around **out** and **out_length** content.
```

Comment:
```suggestion
**s2n_client_hello_get_session_id_length** stores the ClientHello session id length in bytes in **out_length**. The **out_length** can be used to allocate the **out** buffer for the **s2n_client_hello_get_session_id** call.
**s2n_client_hello_get_session_id** copies up to **max_length** bytes of the ClientHello session_id into the **out** buffer and stores the number of copied bytes in **out_length**.
```

We generally don't discuss standard error handling for every method. It's covered in https://github.com/aws/s2n-tls/blob/12720117108630907377eedfe79de7e138c0e107/docs/USAGE-GUIDE.md#error-handling

Reply:
Writing docs is hard (: Thanks, updated.